### PR TITLE
[INLONG-8385][DataProxy] Add take method in BufferQueue class

### LINKED ISSUE #8385: [Improve][DataProxy] Add take method in BufferQueue class
### Description

Currently there is only the poll method, and the take method needs to be added according to the scene

### InLong Component

InLong DataProxy

### Are you willing to submit PR?

- [X] Yes, I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://www.apache.org/foundation/policies/conduct)

### PR DESCRIPTION
**Pull request by gosonzhang**
- Fixes #8385

1. Add takeDispatchedRecord() in BufferQueue class
2. Replace calling pollDispatchedRecord() with takeDispatchedRecord() in the MessageQueueZoneWorker class

### CODE REVIEW

**Review by baomingyu — APPROVED** [review]
+1

**Review by luchunliang — APPROVED** [review]

File: inlong-dataproxy/dataproxy-source/src/main/java/org/apache/inlong/dataproxy/utils/BufferQueue.java [hunk 1]
```diff
@@ -61,6 +61,19 @@ public A pollRecord() {
         return record;
     }
 
+    /**
+     * Take record
+     */
+    public A takeRecord() {
+        try {
+            A record = queue.take();
+            this.offerCount.incrementAndGet();
```
Comment (luchunliang) [inline inlong-dataproxy/dataproxy-source/src/main/java/org/apache/inlong/dataproxy/utils/BufferQueue.java:70]: It is take operation, need to invoke
this.pollCount.getAndIncrement();

File: inlong-dataproxy/dataproxy-source/src/main/java/org/apache/inlong/dataproxy/utils/BufferQueue.java [hunk 2]
```diff
@@ -61,6 +61,19 @@ public A pollRecord() {
         return record;
     }
 
+    /**
+     * Take record
+     */
+    public A takeRecord() {
+        try {
+            A record = queue.take();
```
Comment (luchunliang) [inline inlong-dataproxy/dataproxy-source/src/main/java/org/apache/inlong/dataproxy/utils/BufferQueue.java:69]: take operation will lock current thread.
Please confirm this feature.

File: inlong-dataproxy/dataproxy-source/src/main/java/org/apache/inlong/dataproxy/utils/BufferQueue.java (hunk 1 above)
Comment (gosonzhang) [reply inlong-dataproxy/dataproxy-source/src/main/java/org/apache/inlong/dataproxy/utils/BufferQueue.java]: done

File: inlong-dataproxy/dataproxy-source/src/main/java/org/apache/inlong/dataproxy/utils/BufferQueue.java (hunk 2 above)
Comment (gosonzhang) [reply inlong-dataproxy/dataproxy-source/src/main/java/org/apache/inlong/dataproxy/utils/BufferQueue.java:70]: It's the take() method specially used, mainly to block until there is data in the queue before proceeding with the next processing

**Pull request merged**